src/main/java/org/folio/service/exchangerate/ExchangeRateSourceService.java

```diff
@@ -0,0 +1,92 @@
+package org.folio.service.exchangerate;
+
+import static javax.ws.rs.core.Response.Status.CONFLICT;
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+
+import java.util.Objects;
+
+import org.apache.commons.lang3.StringUtils;
+import org.folio.dao.exchangerate.ExchangeRateSourceDAO;
+import org.folio.rest.core.model.RequestContext;
+import org.folio.rest.exception.HttpException;
+import org.folio.rest.jaxrs.model.ExchangeRateSource;
+import org.folio.rest.util.ErrorCodes;
+
+import io.vertx.core.Future;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@AllArgsConstructor
+public class ExchangeRateSourceService {
+
+  private ExchangeRateSourceDAO exchangeRateSourceDAO;
+
+  public Future<ExchangeRateSource> getExchangeRateSource(RequestContext requestContext) {
+    var dbClient = requestContext.toDBClient();
+    return dbClient.withTrans(conn -> exchangeRateSourceDAO.getExchangeRateSource(conn))
```

Comment:
it is just get operation, openning transaction is not required here, can user dbClient.withConn